src/bin/cql-stress-cassandra-stress/settings/command/mod.rs

```diff
@@ -142,17 +142,31 @@ impl CommandParams {
             .query(schema.construct_keyspace_creation_query(), ())
             .await?;
         session.use_keyspace(&schema.keyspace, true).await?;
+
+        if self.counter.is_some() {
+            session
+                .query(
+                    schema.construct_counter_table_creation_query(&column.columns),
+                    (),
+                )
+                .await
+                .context("Failed to create counter table")?;
+
+            // When `mixed` is Some, `counter` is Some as well.
+            // This is because `mixed` command can run both regular and counter operations.
+            if self.mixed.is_some() {
+                session
+                    .query(schema.construct_table_creation_query(&column.columns), ())
+                    .await
+                    .context("Failed to create standard table")?;
+            }
+            return Ok(());
+        }
+
         session
             .query(schema.construct_table_creation_query(&column.columns), ())
             .await
             .context("Failed to create standard table")?;
-        session
-            .query(
-                schema.construct_counter_table_creation_query(&column.columns),
-                (),
-            )
-            .await
-            .context("Failed to create counter table")?;
 
         Ok(())
     }
```

Comment:
I believe the logic would be much more readable if you separated table conditions from table creation, e.g.:
```rust
let (create_standard_table, create_counter_table) = match (self.counter, self.mixed) {
  // conditions
}

// Actual table creation code
if create_standard_table {
...
}

if create_counter_table {
...
}
```

Reply:
I love this! I'll apply the suggestion.